### PR TITLE
CVE-2011-3953 and CVE-2016-1697

### PR DESCRIPTION
**Pull request by JohnPMurray**
Curated CVE-2011-3953 and CVE-2016-1697 according to project round 1 instructions

### CODE REVIEW

File: cves/CVE-2011-3953.yml [hunk 1]
```diff
@@ -156,11 +181,24 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      Defense in depthwas carried out in this area, making the issue less severe.
```
Comment (JoncarloLovesSecurity) [inline cves/CVE-2011-3953.yml]: There should be a space between "depth" and "was" in this sentence

File: cves/CVE-2011-3953.yml [hunk 2]
```diff
@@ -156,11 +181,24 @@ lessons:
...
+      months without being fully fixed.
   least_privilege:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      Although the target site should have had access to the system clipboard
+      in order for the paste operation to be carried out, access to the clipboard
+      was maintained for too long, allowing attackers to violate confidentiality.
+      The eventual fix involved further restricting this priveledge.
```
Comment (JoncarloLovesSecurity) [inline cves/CVE-2011-3953.yml]: Privilege is misspelled in this sentence.

File: cves/CVE-2011-3953.yml [hunk 3]
```diff
@@ -200,4 +243,26 @@ mistakes:
...
+    for quite some time.
+    The developers did a good job during the design of the product by applying
+    defense in depth to this area, so that users were at least notified when
+    something was wrong. They could have possibly gone further than simply
+    notifying the users, but this was likely a design tradeoff that the
+    developers knew about going in - allowing sites to remain in infinite
+    javascript loops in order to prevent a total crash.
+    The mitigation
+    for this CWE states that priveledges should be extended only to
+    complete the exact task at hand. The fix for this issue does not
```
Comment (JoncarloLovesSecurity) [inline cves/CVE-2011-3953.yml]: There is an unnecessary newline after mitigation on line 257 and privileges is misspelled on line 258.

File: cves/CVE-2011-3953.yml [hunk 4]
```diff
@@ -82,9 +91,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    There didnt seem to be any unit tests involved with this vulnerability.
+    Testing seems to have been done by hand once the vulnerability was found.
```
Comment (JoncarloLovesSecurity) [inline cves/CVE-2011-3953.yml]: "Didnt" should be rewritten as "did not" on line 95.

File: cves/CVE-2011-3953.yml [hunk 5]
```diff
@@ -32,7 +32,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  After a user performed a paste operation, an attacker could start a loop that
+  would continiuously monitor the users clipboard. This could allow attackers 
+  to steal password information and other sensitive data from the clipboard
```
Comment (JoncarloLovesSecurity) [inline cves/CVE-2011-3953.yml]: Perhaps you should specify in the description that the "attacker" in this scenario would a malicious page you are visiting in the Chrome browser.

File: cves/CVE-2016-1697.yml [hunk 6]
```diff
@@ -32,7 +32,12 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  When loading a frame, the chromium rendering engine failed to disable frame
+  navigations, allowing attackers create javascript code that would navigate
+  to access a different page from an untrusted origin and perform cross site 
+  scripting
```
Comment (JoncarloLovesSecurity) [inline cves/CVE-2016-1697.yml]: I think there's a lot here to could be simplified to less technical terms. Specifically I would try to explain what a frame navigation is or maybe just focus on the fact that this vulnerability allowed malicious websites to access the data on other sites you were visiting at the time.

File: cves/CVE-2016-1697.yml [hunk 7]
```diff
@@ -128,12 +151,12 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: |
+    There was a big rush to get this bug fixed before the next stable launch,
+    which may have affected the fix that was implemented.
```
Comment (JoncarloLovesSecurity) [inline cves/CVE-2016-1697.yml]: I would change this sentence from commenting on how the big rush _may_ have affected the fix to talking about how the developers were in a rush to introduce this fix and how it affected the decisions they made and how fast the fix was developed.

File: cves/CVE-2016-1697.yml [hunk 8]
```diff
@@ -174,8 +197,13 @@ lessons:
     applies: 
     note: 
   secure_by_default:
-    applies: 
-    note: 
+    applies: True
+    note: |
+      The fix involved disabling a pages ability to navigate between frames
+      during a certain point in the load process. It appears that taking
+      away this priveledge didn't have any negative side effects, meaning
```
Comment (JoncarloLovesSecurity) [inline cves/CVE-2016-1697.yml]: Privilege is misspelled on line 204. I also usually prefer using "did not" to using "didn't".

File: cves/CVE-2016-1697.yml [hunk 9]
```diff
@@ -194,4 +222,18 @@ mistakes:
...
+    choice would have been to have them disabled by default. 
+    The suggested mitigation for this type of cross site scripting is minimize
+    the attack surface. Disabling frame navigatons while detatching a frame
+    completely cuts out the portion of the attack surface that is exploited
+    by this vulnerability. The fix completely mitigates this exploit, since 
+    frame navigations cannot be exploited if they are entirely disabled. 
+    This change, however, was only applied to one part of the code. It may be
+    worthwhile to review other similar areas of the attack surface and determine
+    whether security by default can be applied to these areas of the code to 
+    prevent similar exploits from happening in the future.
```
Comment (JoncarloLovesSecurity) [inline cves/CVE-2016-1697.yml]: This is good already but I feel like this CVE report lacked an explanation of what Same Origin Policy is and why it needs to be enforced in the Chrome browser. I would add an explanation of Same Origin Policy and explain why being able to bypass it is a big security risk.

File: cves/CVE-2011-3953.yml (hunk 1 above)
Comment (JohnPMurray) [reply cves/CVE-2011-3953.yml]: fixed

File: cves/CVE-2011-3953.yml (hunk 2 above)
Comment (JohnPMurray) [reply cves/CVE-2011-3953.yml]: fixed all instances of the word "privilege" on this cve

File: cves/CVE-2011-3953.yml (hunk 3 above)
Comment (JohnPMurray) [reply cves/CVE-2011-3953.yml]: fixed

File: cves/CVE-2011-3953.yml (hunk 4 above)
Comment (JohnPMurray) [reply cves/CVE-2011-3953.yml]: fixed

File: cves/CVE-2011-3953.yml (hunk 5 above)
Comment (JohnPMurray) [reply cves/CVE-2011-3953.yml]: I changed it to "an attacker can use malicious code on a webpage". I didn't want to specify that the attacker themself created the page, as there could be other ways to execute malicious code through a webpage.

File: cves/CVE-2016-1697.yml (hunk 9 above)
Comment (JohnPMurray) [reply cves/CVE-2016-1697.yml]: I couldn't find a spot where a definition of the same origin policy would belong. I tried to solve this problem in my initial pull request by providing some of the consequences of violating the policy as examples of how the software might be exploited, without directly mentioning the policy. I suppose I could define the policy in the description, but I'd rather keep it simple. Let me know if you disagree, and maybe I can fit in an explanation.

File: cves/CVE-2016-1697.yml (hunk 6 above)
Comment (JohnPMurray) [reply cves/CVE-2016-1697.yml]: I definitely agree. I had some trouble figuring out how to word the description in a less technical way during my initial pull request. I've now edited the description to simplify things, and not get into the specifics of frame navigation.

File: cves/CVE-2016-1697.yml (hunk 7 above)
Comment (JohnPMurray) [reply cves/CVE-2016-1697.yml]: Good idea, it seemed somewhat incomplete the way it was. I've changed the explanation to discuss some of the consequences of the rush.

File: cves/CVE-2016-1697.yml (hunk 8 above)
Comment (JohnPMurray) [reply cves/CVE-2016-1697.yml]: fixed

File: cves/CVE-2011-3953.yml [hunk 10]
```diff
@@ -134,12 +152,20 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: |
+    This bug fix was delayed multiple times, and seemed to be picked back up
+    again at the start of the next milestone, likely due to other deliverables
+    that were deemed to be more important. These are some of the major milestones
+    that delayed the progress of the vulnerability fix.
```
Comment (KylerF) [inline cves/CVE-2011-3953.yml]: You may want to mention that the high level of difficulty in exploiting this vulnerability also attributed to the delays and low priority as discussed in the issue logs.

Comment (JohnPMurray) [reply cves/CVE-2011-3953.yml]: I think mentioning that the deliverables were deemed "more important" than the fix is enough here. The low priority is mentioned elsewhere where it is a little more relevant. Let me know if you disagree.

**Pull request merged**